source-dropbox/main.go

```diff
@@ -0,0 +1,300 @@
+package main
+
+import (
+	"cmp"
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"slices"
+	"time"
+
+	"github.com/dropbox/dropbox-sdk-go-unofficial/v6/dropbox"
+	"github.com/dropbox/dropbox-sdk-go-unofficial/v6/dropbox/files"
+	"github.com/estuary/connectors/filesource"
+	"github.com/estuary/flow/go/parser"
+	pf "github.com/estuary/flow/go/protocols/flow"
+)
+
+type config struct {
+	Path        string         `json:"path"`
+	Credentials *Credentials   `json:"credentials"`
+	Parser      *parser.Config `json:"parser"`
+	MatchKeys   string         `json:"matchKeys,omitempty"`
+	Advanced    advancedConfig `json:"advanced"`
+}
+
+type advancedConfig struct {
+	AscendingKeys bool `json:"ascendingKeys,omitempty"`
+}
+
+func (c config) Validate() error {
+	requiredProperties := [][]string{
+		{"Path", c.Path},
+	}
+	for _, req := range requiredProperties {
+		if req[1] == "" {
+			return fmt.Errorf("missing '%s'", req[0])
+		}
+	}
+	return nil
+}
+
+func (c config) DiscoverRoot() string {
+	return c.Path
+}
+
+func (c config) RecommendedName() string {
+	return c.Path[1:]
+}
+
+func (c config) FilesAreMonotonic() bool {
+	return c.Advanced.AscendingKeys
+}
+
+func (c config) ParserConfig() *parser.Config {
+	return c.Parser
+}
+
+func (c config) PathRegex() string {
+	return c.MatchKeys
+}
+
+func newDropboxStore(ctx context.Context, cfg config) (*dropboxStore, error) {
+	oauthClient := cfg.Credentials.GetOauthClient(ctx)
+	config := dropbox.Config{
+		Client:   oauthClient,
+		LogLevel: dropbox.LogDebug,
```

Comment:
What kind of logs does this produce? Are they useful? Having debug logging enabled seems a little surprising. Without actually looking at what kind of logging we are getting, I would suggest using `LogOff` if the configured connector log level is `Info` or "higher", and only set this to `LogDebug` if the connector log level is `Debug`. Usually we want to have more direct control over logs since they are user-facing, and the SDK logs can be confusing for users.

Reply:
I actually haven't been able to find any logs produced by the SDK. I've tried using both `LogDebug` and `LogInfo` as the `LogLevel`, and providing a `Logger` to the `dropbox.Config`, but I'm not seeing any SDK generated logs show up. I'm leaning toward removing that line since it's not adding any value.

Reply:
Yeah that sounds good